examples/chip-tool/commands/clusters/ClusterCommand.h

```diff
@@ -118,18 +118,23 @@ class ClusterCommand : public InteractionModelCommands, public ModelCommand, pub
 
         if (data != nullptr)
         {
-            LogErrorOnFailure(RemoteDataModelLogger::LogCommandAsJSON(path, data));
-
-            error = DataModelLogger::LogCommand(path, data);
+            {
+                // log a snapshot to not advance the data reader.
+                chip::TLV::TLVReader logTlvReader;
+                logTlvReader.Init(*data);
+                LogErrorOnFailure(RemoteDataModelLogger::LogCommandAsJSON(path, &logTlvReader));
+                error = DataModelLogger::LogCommand(path, &logTlvReader);
```

Comment:
@yunhanw-google Shouldn't the _callee_ have made a snapshot that it then modifies, so it does not modify the passed-in reader?

Reply:
yes, callee has made snapshot, this snapshot would be used for two times, one is here, another is around line 139..counterTlvReader.Init(*data); therefore I decide to snapshot before touching data, leave the tlv copy in data as not-touched.